### PR TITLE
Create core classes

### LINKED ISSUE #6: Create core classes
Create Shop
Create Merchant
Create User
Create Buyer
Create Product
Cart
ItemQuantityList
Main
Tag

### PR DESCRIPTION
**Pull request by Rebecca-Elliott**
## PR type

- [x] Feature
- [ ] Bug Fix
- [ ] Refactor
- [ ] Performance Optimization
- [ ] Documentation Update

## Description

Creates a starting point for the core classes. This is not feature complete and lacks tests. Issues have been created for the missing tests, #7 and #8. Some of the functionality is not quite correct, these bugs are captured in #9, #10, #11, #13, and #14. The next two features that should be developed are #12 and #15.

## Related Issues/Tickets

- Related Issue #
- Closes #6
## QA Instruction, Screenshots etc...

Run main to see a small example of a shop with a product.

## Added/updated tests?

- [ ] Yes
- [x] No: _replace this line with details on why tests are not included in this PR_
Test creation is pending issue#7 and Issue#8

## PR Checklist

- [X] I have rebased this branch against develop/latest
- [X] All the tests are passing
- [X] My commit messages are descriptive and useful

### CODE REVIEW

**Comment by mpdesantis** [comment]
Hi Rebecca,

Thank you for starting out the project! All the classes make sense to me, and your members and methods look very complete.

I can't officially approve this MR, as I'm waiting on a valid invite to the repository, but I would approve if I could :)

@npono21 Are you able to issue me an invite to the repository when you get a chance? Thanks!

**Review by npono21 — APPROVED** [review]
Everything looks good! Awesome!

**Review by ArthurAtangana — APPROVED** [review]

**Pull request merged**
